service/src/main/java/uk/gov/hmcts/reform/fpl/service/ApplicantLocalAuthorityService.java

```diff
@@ -0,0 +1,234 @@
+import uk.gov.hmcts.reform.fpl.model.common.dynamic.DynamicList;
+import uk.gov.hmcts.reform.fpl.model.event.LocalAuthorityEventData;
+import uk.gov.hmcts.reform.fpl.utils.ElementUtils;
+import uk.gov.hmcts.reform.rd.model.Organisation;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.stream.Stream;
+
+import static java.util.Collections.emptyList;
+import static java.util.Objects.nonNull;
+import static java.util.Optional.ofNullable;
+import static java.util.stream.Collectors.toList;
+import static org.apache.commons.lang3.ObjectUtils.firstNonNull;
+import static org.apache.commons.lang3.ObjectUtils.isNotEmpty;
+import static org.springframework.util.ObjectUtils.isEmpty;
+import static uk.gov.hmcts.reform.fpl.enums.ColleagueRole.SOLICITOR;
+import static uk.gov.hmcts.reform.fpl.enums.YesNo.YES;
+import static uk.gov.hmcts.reform.fpl.utils.ElementUtils.addMissingIds;
+import static uk.gov.hmcts.reform.fpl.utils.ElementUtils.asDynamicList;
+import static uk.gov.hmcts.reform.fpl.utils.ElementUtils.element;
+import static uk.gov.hmcts.reform.fpl.utils.ElementUtils.nullSafeList;
+import static uk.gov.hmcts.reform.fpl.utils.ElementUtils.unwrapElements;
+
+@Service
+@RequiredArgsConstructor
+public class ApplicantLocalAuthorityService {
+
+    private final PbaNumberService pbaNumberService;
+    private final OrganisationService organisationService;
+    private final ValidateEmailService validateEmailService;
+
+    public LocalAuthority getUserLocalAuthority(CaseData caseData) {
+
+        if (isEmpty(caseData.getLocalAuthorities())) {
+            return migrateFromLegacyApplicant(caseData).orElseGet(this::getFromOrganisation);
+        } else {
+            return caseData.getLocalAuthorities().get(0).getValue();
+        }
+    }
+
+    public void normalisePba(LocalAuthority localAuthority) {
+
+        localAuthority.setPbaNumber(pbaNumberService.update(localAuthority.getPbaNumber()));
+    }
+
+    public List<String> validateLocalAuthority(LocalAuthority localAuthority) {
+
+        final List<String> errors = new ArrayList<>();
+
+        errors.addAll(pbaNumberService.validate(localAuthority.getPbaNumber()));
+        errors.addAll(validateEmailService.validateIfPresent(localAuthority.getEmail()));
+
+        return errors;
+    }
+
+    public List<String> validateColleagues(List<Element<Colleague>> colleagues) {
+
+        final List<String> colleaguesEmails = unwrapElements(colleagues)
+            .stream()
+            .map(Colleague::getEmail)
+            .collect(toList());
+
+        return validateEmailService.validate(colleaguesEmails, "Colleague");
+    }
+
+    public List<Element<Colleague>> updateMainContact(LocalAuthorityEventData eventData) {
+
+        final List<Element<Colleague>> colleagues = eventData.getLocalAuthorityColleagues();
+
+        if (isEmpty(colleagues)) {
+            return colleagues;
+        }
```

Comment:
Could you elaborate on this please?

Reply:
NPE protection, couple line below there is 
```
colleagues.forEach
```
which must be protected from NPE when colleagues is null